inginious/frontend/plugins/multilang/static/custom_input.js

```diff
@@ -65,35 +65,75 @@ function displayCustomInputResults(data, customTestOutputArea = null, placeholde
     }
 }
 
+function waitForCustomTest(customTestId) {
+    setTimeout(() => {
+        $.get("/api/custom_input_notebook/", {"custom_test_id": customTestId})
+            .done((data) => {
+                data = JSON.parse(data);
+                if ("status" in data && data["status"] === "waiting") {
+                    displayTaskLoadingAlert(data, null);
+                    waitForCustomTest(customTestId);
+                } else if ("status" in data && "result" in data) {
```

Comment:
In the checks for the test status, the type of the "done" status case can be verified as in the "waiting" case. Anyway, it`s already being verified through the result attribute.